accounts/migrations/0010_auto_20170510_0945.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11 on 2017-05-10 09:45
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def replace_empty_email_addresses(apps, schema_editor):
+    User = apps.get_model("auth", "User")
+    users = User.objects.filter(email="")
+    for user in users:
+        user.email = '%s@freesound.org' % user.username
+        user.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('accounts', '0009_sameuser'),
+    ]
+
+    # Add unique constraint to 'email' field in auth_user table
+    operations = [
+        migrations.RunPython(replace_empty_email_addresses, migrations.RunPython.noop),
+        migrations.RunSQL('ALTER TABLE "auth_user" ADD CONSTRAINT "auth_user_email_uniq" UNIQUE ("email");'),
```

Comment:
I made the constraint on upper (https://github.com/MTG/freesound/pull/763/commits/cc6692e89cae7b8ab05f2654ab73b0f797909d76)
There is already an index on email upper.